wait/host_port.go

```diff
@@ -207,10 +215,16 @@ func internalCheck(ctx context.Context, internalPort nat.Port, target StrategyTa
 			return fmt.Errorf("%w, host port waiting failed", err)
 		}
 
+		/*
+			Docker maps exit code 127 to 126, while Podman treats them separately.
+			Handle both to ensure compatibility with Docker and Podman.
+		*/
 		if exitCode == 0 {
 			break
-		} else if exitCode == 126 {
+		} else if exitCode == exitEaccess {
 			return errShellNotExecutable
+		} else if exitCode == exitCmdNotFound {
+			return errShellNotFound
```

Comment:
suggestion: For multiple cases like this a select is easer to follow. The return nil at the end would need to be removed too.

This won't apply cleanly because github doesn't allow suggestions to unedited code:
```suggestion
		select exitCode {
		case 0:
			return nil
		case exitEaccess:
			return errShellNotExecutable
		case exitCmdNotFound:
			return errShellNotFound
		}
```

Reply:
Thanks for pointing out.